### ISSUE TITLE
Echo arena is no longer able to be played with Revive

### ISSUE THREAD

**Issue by DDC667** [LibreVR/Revive#1767]
I have a Reverb G2 with index controllers, the only game I use with revive is ECHO ARENA. until today this was never an issue, sometimes this would happen, but I could fix it by simply starting the game without the controller on, then once loaded I could turn them on and it would begin tracking them. Now no matter what I do it will not load the controllers, it locks them to my headset, and it locks the headsets horizon too, I HAVE DONE EVERYTYHING TO FIX THIS, steam uninstalled wmr uninstalled removed revive, EVERYTHING get the exact same results every time.

I do nothing special when starting revive, I always lunch my wmr first, get loaded into Skyview apartment, then I load SteamVR Witch Starts revive dashboard. I pull up the revive dash board and lunch the game. Used to work flawlessly. Now no matter how I try to start it, it will lock controllers to my headset where I only see my thumbs. 

I make youtube content and cant make any of my echo VR video's now.

**Comment by Godmorgonsverige** [comment]
You could try out the 3.0.0 pre release: https://github.com/LibreVR/Revive/releases/download/3.0.0/ReviveInstaller.exe

**Comment by fearless403** [comment]
After all my research trying to get Demeo to work. It finally did combining everything I saw. I was also able to get Echo Arena to launch too.

- Reinstall Revive
- On Oculus go to Settings > General > OpenXR Runtime (Set Oculus Active)
- In Oculus go to Friends > 3 dots top left > appear offline
- Each time you launch SteamVR you will quit Revive Dashboard then start Revive again via bottom right of desktop window you will see your tasks.
- Right Click on Revive Dashboard > Click "Use OpenVR Runtime"
- Launch the game

Hope this helps

**Comment by DDC667** [comment]
I turned ovr advanced settings, and it works right from SteamVR now, with controllers on. Thanks, issue was not even revive

**Comment by Jlocky404** [comment]
> * Right Click on Revive Dashboard > Click "Use OpenVR Runtime"

Hey so I have a Windows Mixed Reality headset, and I did all that but I'm stuck at this part. Where is exactly is this "Use OpenVR Runtime" option and at that where am I supposed to right click to get that option. Do you mind providing some type of visual instruction? I'm a bit lost by just reading.

**Comment by Jlocky404** [comment]
Thank you so much dude this worked perfectly. Unfortunately, there is an Issue with Windows Mixed Reality headsets, not having support for OpenXR, so I can launch all the games flawlessly with this method, but either get a crash after a while or I'm not able to do anything because it won't detect my controllers.

Gonna look into though, for a possible fix. Hoping for some type of available plugin or maybe another repository with something helpful.





> * Each time you launch SteamVR you will quit Revive Dashboard then start Revive again

**Also this method works perfectly, but this is for future viewers with this issue.** 
Revive typically starts up alongside with SteamVR, so you don't have to exit anything or start it manually. So just go the Desktop's bottom right task drop down menu after you start SteamVR and hit "Use VR Runtime". Reason I'm stating this is because having Revive open before you start SteamVR will cause two (Revive Dashboards) to be open at the same time and will cause a "Another Instance is already running error" when trying to run a game. (Also, won't allow you to close out one without closing the other, which will cause a crash).

**Comment by fearless403** [comment]
Yes Revive launches automatically with SteamVR but what I noticed is when you launch a game your controllers won’t register. Just the simple quit and relaunch of revive works and clicking “Use OpenVR Runtime” because it unchecks itself each launch.  I Have to do that each time I launch SteamVR in  order to play. @Jlocky404

**Comment by Jlocky404** [comment]
You're right that did work, I was able to open up First Contact and play no problems. 

> You can quit revive while steamVR is open.

That crash I got seems to be random thing

I do have some questions I need ask though. After you close Revive Dashboard, does the button for it come back for you in the SteamVR Dashboard? Because after I close Revive, it causes the Revive button on the SteamVR dashboard to disappear and I can't get it back until I close SteamVR. And since I can't load up the games directly from Desktop, I have to load them directly from SteamVR. 

>  I have to load them directly from SteamVR. 
This itself is fine. As I can load up First Contact, and the other pre-loaded Oculus 'games' after I do the close Revive and reopen method (and it also detects my controllers). But I still have one seemingly last problem, Echo Arena. After doing the close-open Revive method, I can load up all the games from SteamVR fine and play except Echo. When Echo tries to load I get to the app but get stuck with these endless 3 loading dots on my screen. 

And when I go to check my desktop I get "An unexpected error has occured while loading"

**Comment by BoxxRoss** [comment]
> * Each time you launch SteamVR you will quit Revive Dashboard then start Revive again via bottom right of desktop window you will see your tasks.
> * Right Click on Revive Dashboard > Click "Use OpenVR Runtime"

this part is confusing me, i cant find any "Use OpenVR Runtime" option. i also dont understand what you mean by "then start Revive again via bottom right of desktop window you will see your tasks." heck is that supposed to mean?

**Comment by fearless403** [comment]
> I do have some questions I need ask though. After you close Revive Dashboard, does the button for it come back for you in the SteamVR Dashboard? Because after I close Revive, it causes the Revive button on the SteamVR dashboard to disappear and I can't get it back until I close SteamVR. And since I can't load up the games directly from Desktop, I have to load them directly from SteamVR. (Revive is still works I just can't see the button anymore)

Yes it does come back for me. When I quit Revive the icon in steamVR disappears but when I relaunch revive it, it reappears. You should be able to load games via steamVR Revive button or by the Revive library on the desktop. 

- When you go to tasks and right click on Revive it will say "Open Library" and you should see this. Attached the image. 
![revive](https://user-images.githubusercontent.com/98560936/164575734-fc5f1428-e284-4486-a3a4-4823e672de83.jpg)

> And when I go to check my desktop I get **_"An unexpected error has occurred while loading"_**. Which isn't very helpful for determining what the problem is. So I tried turning back on Oculus's OpenXR Runtime as the active Runtime, but this made it 

Not quite sure how to help with that. Best thing I could say is try this

- Turn off anti-virus temporarily just to see if game launches. If not then enable it again.
- Check Graphic card drivers are up to date if using Nvidia use Geforce Experience if AMD use AMD Adrenaline Software
- Find the game file > right click > properties > compatibility > Run this program as an administrator
- Back in Revive try launching via library or right click it in tasks and "inject" it.

While trying each step just close revive and relaunch it and don't forget to click OpenVR Runtime. Never hurts to try.

**Comment by Jlocky404** [comment]
> this part is confusing me, i cant find any "Use OpenVR Runtime" option. i also dont understand what you mean by "then start Revive again via bottom right of desktop window you will see your tasks." heck is that supposed to mean?
![InkedScreenshot_171_LI](https://user-images.githubusercontent.com/103242693/164577337-6907facd-f950-4be3-b175-bfca2c47e3b6.jpg)
![Screenshot_172](https://user-images.githubusercontent.com/103242693/164577344-bdc8af1e-1415-4fee-a5d4-6afc2933a437.png)
![Screenshot_173](https://user-images.githubusercontent.com/103242693/164577354-cfc1990e-302b-47bc-b2df-07b21ac39eb2.png)

**Comment by fearless403** [comment]
> this part is confusing me, i cant find any "Use OpenVR Runtime" option.

Seeing your screenshots, I see you found it.

> i also dont understand what you mean by "then start Revive again via bottom right of desktop window you will see your tasks." heck is that supposed to mean?

Sorry about that i worded it wrong. I edited that part what I meant to say was quit revive via the tasks menu at the bottom right of your screen. Then relaunch revive.
